### DISCUSSION TITLE
Discussion: partial methods and interfaces

### DISCUSSION THREAD

**Discussion by alrz** [dotnet/csharplang#4168] [Language Ideas]
Currently, all partial methods require a body but in an interface, you might still want to keep it abstract:
```cs
partial interface I { [A] public partial void M(); }
partial interface I { [B] public partial void M(); }
```
Equilvanet to:
```cs
partial interface I { [A, B] public void M(); }
```
Note that `partial void M();` follows the old semantic and optionally emits `private void M() { }` if it has an implementation part.

**Comment by YairHalberstadt** [comment]
This should also be the case for abstract methods in classes.

**Reply by alrz** [reply]
I think the main use case would be merged attributes, perhaps it could be considered alongside https://github.com/dotnet/csharplang/issues/3658 cc @RikkiGibson 

PS: this is the behavior I'd expect with fields etc if we ever go for "partial everything" support.

**Reply by RikkiGibson** [reply]
It feels like the suggestion is to decide whether a partial method with access modifiers is a DIM based on presence of an implementation part.

It would help if you could share some concrete use cases for this.

**Reply by alrz** [reply]
@RikkiGibson 

The one in the opening post is my use case. I want to add some attributes to interface methods in separate partial decls.

Note the result would not use DIM in any ways, I only added `public` to use new semantics for `partial`.

**Reply by alrz** [reply]
>new semantics for partial.

However that itself requires a body which I don't want to add. 

On the other hand, `partial void M()` would actually emit a "private method" (if it has a implementation part) which was a little surprising.

**Reply by RikkiGibson** [reply]
Sorry, I was trying to ask what kinds of applications or features are you having difficulty implementing because of this limitation of the partial methods design.

**Reply by alrz** [reply]
Nothing fancy, just wanted to add some framework-specific attributes to an interface I linked from another project.

I did manage to wokaround this though, using placeholder attributes where they were not available.
